fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/rule/SingleTableRewriteBaseRule.java

```diff
@@ -0,0 +1,70 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.rule.transformation.materialization.rule;
+
+import com.google.common.collect.Lists;
+import com.starrocks.sql.optimizer.ExpressionContext;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.rule.RuleType;
+import com.starrocks.sql.optimizer.statistics.Statistics;
+import com.starrocks.sql.optimizer.statistics.StatisticsCalculator;
+
+import java.util.Collections;
+import java.util.List;
+
+public abstract class SingleTableRewriteBaseRule extends BaseMaterializedViewRewriteRule {
+    public SingleTableRewriteBaseRule(RuleType type, Pattern pattern) {
+        super(type, pattern);
+    }
+
+    // select OptExpression based on statistics
+    @Override
+    public List<OptExpression> transform(OptExpression queryExpression, OptimizerContext context) {
+        List<OptExpression> expressions = super.transform(queryExpression, context);
+        if (expressions == null || expressions.isEmpty()) {
+            return Lists.newArrayList();
+        } else {
+            if (expressions.size() == 1) {
+                return expressions;
+            }
+            // compute the statistics of OptExpression
+            for (OptExpression expression : expressions) {
+                calculateStatistics(expression, context);
+            }
+            // sort expressions based on statistics output row count
+            Collections.sort(expressions, (expression1, expression2) -> {
+                if (expression1.getStatistics() == null && expression1.getStatistics() == null) {
```

Comment:
```suggestion
                if (expression1.getStatistics() == null && expression1.getStatistics() == null) {
```
all expression1 ??????

Reply:
fixed